spec/setup/helpers/perf_helper_spec.rb

```diff
@@ -41,13 +75,217 @@ def options
       hosts[0]['platform'] = Beaker::Platform.new('el-6-x86_64')
       allow(subject).to receive(:hosts).and_return(hosts)
       allow(subject).to receive(:agents).and_return(hosts)
-      expect(subject.hosts[0]).to receive(:install_package).with("https://dl.fedoraproject.org/pub/epel/epel-release-latest-6.noarch.rpm", '', nil, :acceptable_exit_codes => [0,1])
+      expect(subject.hosts[0]).to receive(:install_package).with("https://dl.fedoraproject.org/pub/epel/epel-release-latest-6.noarch.rpm", '', nil, :acceptable_exit_codes => [0, 1])
       allow(subject).to receive(:on)
       expect(subject).to receive(:on).with(hosts[0], 'chkconfig iptables off').once
       subject.install_epel_packages
     end
 
   end
 
+  context '.setup_r10k' do
+
+    # TODO: is this the best way to handle masters?
+    let!(:masters) {[{'platform' => Beaker::Platform.new('centos-6.5-x86_64')}]}
+
+    it 'adds the private key, configures ssh, and installs git on each master' do
+
+      allow(subject).to receive(:masters).and_return(masters)
+      expect(subject).to receive(:select_hosts).with({:roles => ['master', 'compile_master']}).and_return(masters)
+
+      # TODO: should these use the private_key and ssh_config variables or are these matchers sufficient?
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/id_rsa", /PRIVATE/ )
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/config", /StrictHostKeyChecking/ )
+
+      expect(subject).to receive(:on).with(masters[0], "chmod 600 /root/.ssh/id_rsa /root/.ssh/config" )
+      expect(subject).to receive(:install_package).with(masters[0], 'git')
+      subject.setup_r10k
+
+    end
+
+  end
+
+  describe '#cloud_config' do
+    let(:host) {:hosts[0]}
+
+    context 'when the cloud.cfg file exists' do
+      it 'ensures preserve_hostname is set to true' do
+        expect(host).to receive(:file_exist?).with(("/etc/cloud/cloud.cfg")).and_return(true)
+        expect(subject).to receive(:on).with(host, "if grep 'preserve_hostname: true' /etc/cloud/cloud.cfg ; then echo 'already set' ; else echo 'preserve_hostname: true' >> /etc/cloud/cloud.cfg ; fi").once
+        subject.cloud_config(host)
+      end
+    end
+
+    context 'when the cloud.cfg file does not exist' do
+      it 'does nothing' do
+        expect(host).to receive(:file_exist?).with(("/etc/cloud/cloud.cfg")).and_return(false)
+        expect(subject).to_not receive(:on)
+        subject.cloud_config(host)
+      end
+    end
+
+  end
+
+  describe '#ec2_workarounds' do
```

Comment:
Sadly, I just deleted this method too, since it is no longer necessary with the new AWS account.